solutions/noisg-19-feast.mdx

```diff
@@ -0,0 +1,83 @@
+---
+id: noisg-19-feast
+source: NOI
+title: 2019 - Feast
+author: Dong Liu
+---
+
+**Time Complexity:** $\mathcal{O}(N\log{\sum a[i]})$.
+
+<LanguageSection>
+
+<CPPSection>
+
+Like a normal Alien Trick problem, we're going to binary search on the maximum $\texttt{penalty}$.
```

Comment:
Also, rather use lambda instead of "penalty"

Reply:
You should probably rather say something like "binary search on the maximum $\lambda$ **where \<condition\>...**" for clarity

Reply:
haven't read the problem but wouldn't penalty be a bit more intuitive? in an equation you can use p for it or smth

Reply:
Lambda is almost always used when talking about Lagrangian relaxation, so I think we should also use it to be consistent